File: test/aidbox_sdk/snapshots/typescript/hl7-fhir-r4-core/RelatedPerson.ts
import { Address } from "./Address";
import { Attachment } from "./Attachment";
import { Period } from "./Period";
import { CodeableConcept } from "./CodeableConcept";
import { ContactPoint } from "./ContactPoint";
import { HumanName } from "./HumanName";
import { DomainResource } from "./DomainResource";
import { Reference } from "./Reference";
import { Identifier } from "./Identifier";
import { BackboneElement } from "./BackboneElement";

export type RelatedPersonCommunication = BackboneElement & {
    language: CodeableConcept;
    preferred?: boolean;
};

export type RelatedPerson = DomainResource & {
    patient: Reference;
    address?: Address[];
    name?: HumanName[];
    birthDate?: string;
    relationship?: CodeableConcept[];
    photo?: Attachment[];
    active?: boolean;
    communication?: RelatedPersonCommunication[];
    identifier?: Identifier[];
    telecom?: ContactPoint[];
    gender?: string;
    period?: Period;
};

File: test/aidbox_sdk/snapshots/python/hl7-fhir-r4-core/RelatedPerson.py
from typing import Optional, List
from base import Address
from base import Attachment
from base import Period
from base import CodeableConcept
from base import ContactPoint
from base import HumanName
from base import DomainResource
from base import Reference
from base import Identifier
from base import BackboneElement

class RelatedPerson_Communication(BackboneElement):
    language: CodeableConcept
    preferred: Optional[bool] = None

class RelatedPerson(DomainResource):
    patient: Reference
    address: Optional[List[Address]] = None
    name: Optional[List[HumanName]] = None
    birth_date: Optional[str] = None
    relationship: Optional[List[CodeableConcept]] = None
    photo: Optional[List[Attachment]] = None
    active: Optional[bool] = None
    communication: Optional[List[RelatedPerson_Communication]] = None
    identifier: Optional[List[Identifier]] = None
    telecom: Optional[List[ContactPoint]] = None
    gender: Optional[str] = None
    period: Optional[Period] = None